### PR TITLE
verbose update of dependencies

### PR DESCRIPTION
**Pull request by debris**
In some cases (#5530) cargo does not print update messages. This pr tries to address this problem.

In a case described in issue #5530, cargo will print additional line:

```diff
    Updating registry `https://github.com/rust-lang/crates.io-index`
+   Updating bitflags v0.9.1 -> v1.0.3
```

If accepted, I will add some tests

### CODE REVIEW

File: src/cargo/ops/cargo_generate_lockfile.rs [hunk 1]
```diff
@@ -208,6 +271,44 @@ pub fn update_lockfile(ws: &Workspace, opts: &UpdateOptions) -> CargoResult<()>
...
         debug!("{:#?}", changes);
 
-        changes.into_iter().map(|(_, v)| v).collect()
+        let mut partly_added = BTreeMap::new();
+        let mut partly_updated = BTreeMap::new();
+        let mut partly_removed = BTreeMap::new();
+
+        {
+            let dependents_iter = changes.iter()
+                .filter(|(_, v)| v.0.len() == 0 && v.1.len() == 0)
```
Comment (debris) [inline src/cargo/ops/cargo_generate_lockfile.rs]: there is one edge case where this check is not enough, e.g.

1. Set up a workspace with the following members and dependencies:

```
a → bitflags="0.9"
b → bitflags="1.0"
c → bitflags="0.9"
```

2. `cargo update` to build the lock file
3. New version of bitflags is release on crates.io - `1.x`
4. Edit `c`'s version of bitflags to `1.0`
5. `cargo update` will print only the message describing an update from `1.0` to `1.x`

**Comment by debris** [comment]
In the second commit, I completely refactored the way diffs are generated. Yet, the output is consistent with the output cargo produces currently. 

diff of `cargo update` run with `cargo 1.26.0 (0e7c5a931 2018-04-06)` and this branch on [parity/master](https://github.com/paritytech/parity) commit `447950e7c160b8ee2c886d3659007fc7cf50567f`

https://www.diffchecker.com/YIRXmxLq

**Comment by alexcrichton** [comment]
Thanks for the PR! I definitely think that #5530 is a good issue to fix in that `cargo update` with no output implies to me that nothing happened nowadays, so if something happens we should print at least *something*!

Mind adding a few tests for the fixed cases here as well? Other than that looks good to go to me!

**Comment by debris** [comment]
sure, I'll try to finish this in the upcoming week

**Comment by alexcrichton** [comment]
Ok I'm gonna close this for now, but @debris if you get a chance to update I can certainly reopen!

**Pull request closed without merging**